### PR TITLE
Check if not a left-click, then bail out of onStart

### PR DESCRIPTION
**Pull request by AlexVvx**
Related to [#248](https://github.com/plotly/plotly.js/issues/248)

This PR adds check in dragElement. On start, if not a left-click, then bail out of onStart. The purpose of this check is to let developers to use their own context menu instead of browser context menu. Without that check there were no possibility to see 'contextmenu' event when click inside plot container.

### CODE REVIEW

File: test/jasmine/tests/dragelement_test.js [hunk 1]
```diff
@@ -122,6 +122,19 @@ describe('dragElement', function() {
...
 
+    it('should not add a cover slip div to the DOM when right click', function() {
+        function countCoverSlip() {
+            return d3.selectAll('.dragcover').size();
+        }
+
+        var options = { element: this.element, gd: this.gd };
+        dragElement.init(options);
+
+        options.element.onmousedown({button: 2})
```
Comment (AlexVvx) [inline test/jasmine/tests/dragelement_test.js]: Not sure, that it's a best way to emulate right click, however I can't see any other options.

Comment (alexcjohnson) [reply test/jasmine/tests/dragelement_test.js]: looks like in [mouse_event](https://github.com/plotly/plotly.js/blob/f1faa2871dbf182847579166af47d1cc2a983707/test/jasmine/assets/mouse_event.js#L11) we have `buttons` but if we add `button` the same way it should allow right-click simulation?

**Comment by AlexVvx** [comment]
I wasn't able to find 'Buttons' in ie spec, but it works well in ie9, so I changed everything to 'buttons'.

File: test/jasmine/tests/dragelement_test.js [hunk 2]
```diff
@@ -122,6 +122,18 @@ describe('dragElement', function() {
...
+    it('should not add a cover slip div to the DOM when right click', function() {
+        function countCoverSlip() {
+            return d3.selectAll('.dragcover').size();
+        }
+
+        var options = { element: this.element, gd: this.gd };
+        dragElement.init(options);
+
+        mouseEvent('mousedown', this.x, this.y, { buttons: 2 });
+        expect(countCoverSlip()).toEqual(0);
```
Comment (alexcjohnson) [inline test/jasmine/tests/dragelement_test.js]: Great! Very clear correspondence to the test above. Two last changes I would suggest, then I think this is ready to go:
- 🌴  Can you move `countCoverSlip` out one scope so we don't have to repeat it?
- Can you continue through `mouseup` as in the test below, and verify that `handleClick` has not been called? (that one ironically looks annoying to DRY 🌴  so you can just repeat the pieces you need.)

Comment (AlexVvx) [reply test/jasmine/tests/dragelement_test.js]: Sure

**Comment by alexcjohnson** [comment]
Fantastic @AlexVvx - thanks for the PR and updates! 💃  - we're calling this a bug fix, so it can go in the release we're planning later today.

**Pull request merged**